eZ/Publish/Core/Repository/Values/User/User.php

```diff
@@ -140,18 +140,19 @@ public function __get($property)
                 return $this->getVersionInfo()->getContentInfo();
 
             case 'id':
-                $versionInfo = $this->getVersionInfo();
-                if (empty($versionInfo)) {
-                    return null;
-                }
-
-                return $versionInfo->getContentInfo()->id;
+                return $this->getVersionInfo()->getContentInfo()->id;
 
             case 'versionInfo':
                 return $this->getVersionInfo();
 
             case 'fields':
                 return $this->getFields();
+
+            case 'content':
+                @trigger_error(
+                    sprintf('%s is and internal property. User itself exposes everything needed.', $property),
+                    E_USER_DEPRECATED
+                );
```

Comment:
Nitpick: as this `trigger_error` does not stop execution, after this line there should be a `break;` (easy to forget when adding next `case` in the future). However this does not change anything right now, so only nitpicking :)